### DISCUSSION TITLE
Install     joplin-vieweb only with a webserver

### DISCUSSION THREAD

**Discussion by Moosbueffel** [joplin-vieweb/joplin-vieweb#78] [Q&A] [answered]
Sorry for my silly questions, but english is not my home language and I'm not really competent in server and networking installations.

Can you explain, how to configure the .yml file for only installing your joplin-vieweb and an webServer, so I can use it? Or do I missunderstood your explanation and it is enough to use the [docker-compose-joplin-only.yml](https://github.com/joplin-vieweb/joplin-vieweb/blob/main/docker-compose-joplin-only.yml).

I have a raspberry and only want to use it locally from my devices at home. So I don't need a reverse proxy or cert. I have some other server application, which run perfect without this (e.g. paperless).

Second question: Which IP is nessesary in Origin? Is this the IP of my raspberry, or as I understand the ip of the devices which should have access to the server app?

**Comment by gri38** [comment, marked as answer]
Hello @Moosbueffel , sorry for the delay, I was not here this week end.
Indeed you just need to run docker-compose on the [docker-compose-joplin-only.yml](https://github.com/joplin-vieweb/joplin-vieweb/blob/main/docker-compose-joplin-only.yml).

In ORIGINS, you must set the address of your raspberry, the IP and/or the hostname you would use to reach it.
In fact, if you input http://my_raspberry/joplin in you webbrowser, or http://192.168.1.2/joplin, then the ORIGIN must be set to:
```yaml
x-common-variables: &common-variables
   ORIGINS: "'http://my_raspberry', 'http://192.168.1.2'"
   JOPLIN_LOGIN_REQUIRED: True
```

Don't forget to set, in the docker-compose-joplin-only.yml, the <PORT-TO-YOUR-PROXY> (you probably want it to be 80)
```yaml
    ports:
      - 80:8000
```

Finally, rename docker-compose-joplin-only.yml to docker-compose.yml, so it's simpler in you docker compose command:
```bash
docker compose up -d
```

Good luck.

**Reply by Moosbueffel** [reply]
Many thanks for this well explanation. It works, but I had a problem, I will describe this for others:

In the ORIGINS I have to add the port e.g. 
`ORIGINS: "'http://mySHS:8008', 'http://192.168.178.51:8008'"`

I assume this is, because I have to use a port different form port 80.

**Reply by gri38** [reply]
thanks for the precision.

**Comment by Moosbueffel** [comment]
A little bit late, but many thanks for your detailed description. It works fine now!

Georg

01.12.2024 17:10:59 Nicolas Guillot ***@***.***>:
